client/cli/src/lib.rs

```diff
@@ -229,78 +227,81 @@ pub trait SubstrateCli: Sized {
 }
 
 /// Initialize the logger
-pub fn init_logger(pattern: &str) {
-	use ansi_term::Colour;
-
-	let mut builder = env_logger::Builder::new();
-	// Disable info logging by default for some modules:
-	builder.filter(Some("ws"), log::LevelFilter::Off);
-	builder.filter(Some("yamux"), log::LevelFilter::Off);
-	builder.filter(Some("cranelift_codegen"), log::LevelFilter::Off);
-	builder.filter(Some("hyper"), log::LevelFilter::Warn);
-	builder.filter(Some("cranelift_wasm"), log::LevelFilter::Warn);
-	// Always log the special target `sc_tracing`, overrides global level
-	builder.filter(Some("sc_tracing"), log::LevelFilter::Trace);
-	// Enable info for others.
-	builder.filter(None, log::LevelFilter::Info);
+pub fn init_logger(
+	pattern: &str,
+	tracing_receiver: sc_tracing::TracingReceiver,
+	tracing_targets: Option<String>
+) {
+	match tracing_log::LogTracer::init() {
+		Ok(_) => (),
+		Err(_) => log::info!(
+			"💬 Not registering Substrate logger, as there is already a global logger registered!"
+		),
+	}
+
+	let mut env_filter = tracing_subscriber::EnvFilter::default()
+		// Disable info logging by default for some modules.
+		.add_directive("ws=off".parse().expect("provided directive is valid"))
+		.add_directive("yamux=off".parse().expect("provided directive is valid"))
+		.add_directive("cranelift_codegen=off".parse().expect("provided directive is valid"))
+		// Set warn logging by default for some modules.
+		.add_directive("cranelife_wasm=warn".parse().expect("provided directive is valid"))
+		.add_directive("hyper=warn".parse().expect("provided directive is valid"))
+		// Always log the special target `sc_tracing`, overrides global level.
+		.add_directive("sc_tracing=trace".parse().expect("provided directive is valid"))
+		// Enable info for others.
+		.add_directive(tracing_subscriber::filter::LevelFilter::INFO.into());
 
 	if let Ok(lvl) = std::env::var("RUST_LOG") {
-		builder.parse_filters(&lvl);
+		if lvl != "" {
+			match lvl.parse() {
+				Ok(directive) => {
+					env_filter = env_filter.add_directive(directive);
+				}
+				// We're not sure if log or tracing is available at this moment, so silently ignore the
+				// parse error.
+				Err(_) => (),
+			}
```

Comment:
Could just be an `if let`

Reply:
Hmm. Like it's written it's very clear why we're not handling the error.

Reply:
The comment can also just be moved above the `if let`